app-shell/src/protocol-storage/file-system.ts

```
@@ -159,7 +165,12 @@ export function analyzeProtocolByKey(
     PROTOCOL_ANALYSIS_DIRECTORY_NAME
   )
   const destFilePath = makeAnalysisFilePath(analysisDirPath)
-  return analyzeProtocolSource(srcDirPath, destFilePath)
+  return readFilesWithinDirectory(srcDirPath).then(protocolFiles => {
+    if (protocolFiles.length === 0) {
+      throw new Error('No valid protocol files found')
+    }
+    return analyzeProtocolSource(protocolFiles[0], destFilePath)
```

I think the behavior of returning the first valid dir is a bit surprising. We should probably throw an error if the length > 1

---

hmm, what i heard from a fas member, some customers keep storing the copied files in src folder/dir so throw an error if length >1 might be unexpected to them.

---

Gotcha, that's good context, thank you. We should probably aim to maintain current behavior then. If multiple files are stored in the src folder, what happens currently when a user analyzes? As long as we are keeping that behavior, I think we're good!

---

@ecormany do you have any thoughts on this?
technically, what Jamey suggested is safer than passing the first file.